pkg/kubelet/pod_workers_test.go

```diff
@@ -2382,3 +2382,52 @@ func Test_allowPodStart(t *testing.T) {
 		})
 	}
 }
+
+func Test_calculateEffectiveGracePeriod(t *testing.T) {
+	zero := int64(0)
+	two := int64(2)
+	five := int64(5)
+	thirty := int64(30)
+	// no overrides, use what's on the spec
+	pod := newNamedPod("1", "ns", "running-pod", false)
```

Comment:
Reading this, I see five tests in one. Could you refactor this to be in a table?

Reply:
Something like https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/pod_workers_test.go#L1691?